### PR TITLE
Finalize Hashbang changes - fix popstate - fix links - fix subdirectory usage

### PR DESCRIPTION
**Pull request by JonathonDunford**
This commit fixes several issues.

This commit adds HistoryPushstates
Adds popstate listener
Adds support for forward/back buttons
Prevents default event from <a> clicks, just use JS
Supports subdirectories by changing ALL routing to pushstates

Now we ware Google friendly :1st_place_medal:

### CODE REVIEW

File: js/main-unminified.js [hunk 1]
```diff
@@ -125340,10 +125340,20 @@
...
-                                                                _this2.self.selectedBase.name
+                                                                _this2.self.selectedBase.name,
+                                                            onClick: function(e) {
+																e.preventDefault();
+																_this2.selectToken.bind(
+																	_this2.self,
+																	token.addr
+ 																);
+																
+																_this2.self.selectTokenAndBase(token.addr, _this2.self.selectedBase.addr);
```
Comment (freeatnet) [inline js/main-unminified.js]: We should probably just call the version bound to `_this2.self`, ie
```javascript
_this2.selectToken.bind(_this2.self, token.addr)();
```
or 
```javascript
_this2.selectToken.bind(_this2.self)(token.addr);
```

File: js/main-unminified.js [hunk 2]
```diff
@@ -125340,10 +125340,20 @@
...
+																history.pushState(null, null, '/#!/trade/' + (token.name || token.addr) + '-' + _this2.self.selectedBase.name);
```
Comment (freeatnet) [inline js/main-unminified.js]: Do we really need the leading `/` in the URL here? (and in other places)

File: js/main-unminified.js [hunk 3]
```diff
@@ -125948,14 +125958,26 @@
                                                                                                 'a',
                                                                                                 {
                                                                                                     href:
-                                                                                                        '#!/trade/' +
+                                                                                                        '/#!/trade/' +
```
Comment (freeatnet) [inline js/main-unminified.js]: Ditto.

File: js/main-unminified.js (hunk 2 above)
Comment (JonathonDunford) [reply js/main-unminified.js]: In history state? Yes. I believe so. This will make google index the site at the root.

Comment (freeatnet) [reply js/main-unminified.js]: Doesn't it break reloads if forkdelta index.html is located in a subdirectory?

Comment (JonathonDunford) [reply js/main-unminified.js]: it works on subdirectories but it acts as if its on root
and indexes on root
which is as intended IMO

File: js/main-unminified.js (hunk 3 above)
Comment (JonathonDunford) [reply js/main-unminified.js]: it works on subdirectories but it acts as if its on root
and indexes on root
which is as intended IMO

File: js/main-unminified.js (hunk 2 above)
Comment (freeatnet) [reply js/main-unminified.js]: 👍

File: js/main-unminified.js (hunk 1 above)
Comment (JonathonDunford) [reply js/main-unminified.js]: Done.

File: js/main-unminified.js [hunk 4]
```diff
@@ -125340,10 +125340,21 @@
...
-                                                                _this2.self.selectedBase.name
+                                                                _this2.self.selectedBase.name,
+                                                            onClick: function(e) {
+																//prevent <a> clicking and changing page
+																e.preventDefault();
+																//bind select token function then call it on click
+																_this2.selectToken.bind(
+																	_this2.self,
+																	token.addr
+ 																)(token.addr);
```
Comment (freeatnet) [inline js/main-unminified.js]: Only one should include `token.addr`: either `bind` arguments or call-time arguments.

**Pull request merged**